mobile/src/main/java/org/openhab/habdroid/core/NotificationHelper.kt

```diff
@@ -123,8 +123,8 @@ class NotificationHelper constructor(private val context: Context) {
         var iconBitmap: Bitmap? = null
 
         if (message.icon != null) {
-            val connection = ConnectionFactory.cloudConnectionOrNull
-            if (connection != null && context.determineDataUsagePolicy().canDoLargeTransfers) {
+            val connection = ConnectionFactory.primaryCloudConnection?.connection
+            if (connection != null && !context.determineDataUsagePolicy().canDoLargeTransfers) {
```

Comment:
@mueller-ma Was this change intentional? It looks wrong on first impression.

Reply:
The `!` is a rebase mistake by myself. Before it was `!context.dataSaverActive()`.